nixos/modules/services/networking/adguardhome.nix

```diff
@@ -167,8 +167,13 @@ in {
       preStart = optionalString (settings != null) ''
         if    [ -e "$STATE_DIRECTORY/AdGuardHome.yaml" ] \
            && [ "${toString cfg.mutableSettings}" = "1" ]; then
+          # First run a schema_version update on the existing configuration
+          # This ensures that both the new config and the existing one have the same schema_version
+          # Note: --check-config has the side effect of modifying the file at rest!
+          ${lib.getExe pkgs.adguardhome} -c "$STATE_DIRECTORY/AdGuardHome.yaml" --check-config
```

Comment:
Should `pkgs.adguardhome` be `cfg.package` instead?

Reply:
True. Give me a second.